src/axom/quest/interface/inout.cpp

```diff
@@ -428,10 +531,38 @@ int inout_mesh_center_of_mass(double* coords)
 
   SLIC_ERROR_IF(coords == nullptr, "supplied buffer 'coords' is null");
 
-  auto& cmass = s_inoutHelper.getCenterOfMass();
-  cmass.array().to_array(coords);
+  const int dim = inout_get_dimension();
+  int rc = QUEST_INOUT_SUCCESS;
+  switch(dim)
+  {
+  case 2:
+    s_inoutHelper2D.getCenterOfMass().array().to_array(coords);
+    break;
+  case 3:
+    s_inoutHelper3D.getCenterOfMass().array().to_array(coords);
+    break;
+  default:
+    rc = QUEST_INOUT_FAILED;
+    break;
+  }
 
-  return QUEST_INOUT_SUCCESS;
+  return rc;
+}
+
+//------------------------------------------------------------------------------
+
+bool inout_evaluate(double x, double y)
+{
+  if(!inout_initialized())
+  {
+    SLIC_WARNING("quest inout query must be initialized "
```

Comment:
Should this be an error instead?

Reply:
For the inout "C" interface, we opted to have everything use return codes so the end user can decide how to do error handling.